### ISSUE TITLE
Documentation? Websites are allowed to proactively request access to my database? Why?

### ISSUE THREAD

**Issue by scruloose** [keepassxreboot/keepassxc-browser#408]
What are the differences between autofill and pages "requesting access" to entries in my database? If I just logged in using "fill username and password" from the context menu, why is the site requesting access to anything? What are the risks/consequences of allowing? Denying?

## Expected Behavior
Assuming a user who is not a dev and not well-versed in the guts of browser extension APIs, there should be some indication somewhere of why dialog boxes requesting access to KeepassXC database entries keep popping up, what degree of access exactly is being asked for and for what purpose, and what results to expect from allowing or denying it.

## Current Behavior

I installed KeepassXC and KeepassXC-browser, and had no trouble configuring them to talk to each other. Cool. Navigate to a login page with a corresponding URL entry, right-click on the username box, and hit "fill username and password", push the "log in" button on the page, and it works. Also cool. As soon as the site's landing page loads, there's a KeepassXC-browser dialog box indicating that the site is "requesting access" to its entry in my KeepassXC database. Um, what? Why? Shouldn't the site just be using its own cookie at this point? Read-only access or read-write access? If I hit Allow, nothing appears to happen. Likewise if I hit Deny... nothing appears to happen. Some logged-in sites trigger this dialog box on Every. Single. Page. Load. Others don't seem to trigger it at all? WTF is even going on? A web search turns up no documentation anywhere to provide any hint as to what this dialog box does, what triggers its arrival, or on what basis I would choose between the options.

## Possible Solution
KeepassXC-browser should have some documentation of its own, besides the link to the KeepassXC documentation, that gives some explanation of the plugins behavior, preferably including:
- Rationale for website-initiated "requests for access" to exist at all
- What degree of access is permitted by pressing "Allow"
- Under what circumstances the dialog will show up
- How to disable it globally, if possible
- What functionality, if any, is lost by hitting "Deny" and "Remember this choice" (or global disable)
- What relationship, if any, exists between "password field detection", "autofill", and this "website is requesting access" dialog

## Debug info
KeePassXC - 2.3.4
KeePassXC-Browser - 1.3.2
Operating system: Linux
Browser: Firefox

**Comment by droidmonkey** [comment]
Please let us know which site triggers the continuos credential authorization.

**Comment by varjolintu** [comment]
An official documentation is released soon.

Every time you go to a web page the extension detects if there are any login input fields available. If there is, and you have a credentials that match with the site URL, the extension is requesting an access to these credentials. Even if you have connected your database to the extension, an access to read your username and password is not granted automatically (although there's a setting in KeePassXC that overrides this and the request is never shown and access is always granted. It's found from Browser Integration -> Advanced -> Never ask before accessing credentials). Denying all access to the credentials is not possible globally, but you can add a certain site to the extension's Site Preferences settings (from the popup -> Settings).

Pressing "Allow" grants KeePassXC to return the credentials once. Also, using "Deny" will do the same. If you remember the choice, the permission for this site will be not asked again. If you wish to change this behaviour, you must edit your entry in KeePassXC and remove/edit the permissions. With KeePassXC 2.3.4 and earlier this can be found from Advanced page when editing an entry (there's an attribute KeePassXC-Browser Settings). With 2.40 and newer it's located in Properties page, shown as Plugin Data.

Autofill is used when the site has an allowed permission and you wish to fill the login credentials automatically. Autocomplete menu means you can click the username/password field and select the correct credentials from there.

**Comment by scruloose** [comment]
> Please let us know which site triggers the continuos credential authorization.

forum.xda-developers.com

**Comment by scruloose** [comment]
> An official documentation is released soon.

I'm glad to hear it!

> Every time you go to a web page the extension detects if there are any login input fields available. If there is, and you have a credentials that match with the site URL, the extension is requesting an access to these credentials. 

The dialog box might benefit from being reworded. As it stands, it seems to indicate that the request is initiated by the website in question, rather than by the extension.

*****

Other than that, thank you for the clarification about the extension's behaviour and options.

**Comment by varjolintu** [comment]
> forum.xda-developers.com

This behaviour is expected because the login form is included in every page if you are not logged in. The only way to get rid of this annoying is to remember your credentials for this page, so the dialog won't trigger again. Other solution is to disable automatic retrieval of the credentials but then you need to manually fill them for all pages.

**Comment by scruloose** [comment]
> > forum.xda-developers.com
> 
> This behaviour is expected because the login form is included in every page if you are not logged in.

That part makes sense, given your explanation upthread.  The thing that still seems strange to me is that it happens when I **am** logged in. Every single page load, every single "open in new tab" to a link within the site, and even switching to a tab with a logged-in xda-developers session in it.  I have permitted that site's first-party cookie, so there should be no need to actually log in again & again. When I expand the site's user menu by clicking on the little userhead icon in the top-right corner of the page, it shows my logged-in control panel, not the login form.

I have indeed selected Allow and Remember now, just to make it shut up.
